dfusion_rust_core/src/database/graph_reader.rs

```diff
@@ -85,16 +88,60 @@ impl DbInterface for GraphReader {
 
     fn get_orders_of_slot(
         &self,
-        _: &U256,
+        auction_id: &U256,
     ) -> Result<Vec<models::Order>, DatabaseError> {
-        unimplemented!()
+        let order_query = entity_query(
+            "SellOrder", EntityFilter::Equal("auctionId".to_string(), auction_id.to_value()),
+        );
+        Ok(self.reader
+            .find(order_query)
+            .map_err(|e| DatabaseError::chain(ErrorKind::ConnectionError, "Could not execute query", e))?
+            .into_iter()
+            .map(models::Order::from)
+            .collect::<Vec<models::Order>>()
+        )
     }
 
     fn get_standing_orders_of_slot(
         &self,
-        _: &U256,
+        auction_id: &U256,
     ) -> Result<[models::StandingOrder; models::NUM_RESERVED_ACCOUNTS], DatabaseError> {
-        unimplemented!()
+        let mut result = StandingOrder::empty_array();
+        for (reserved_account_id, item) in result.iter_mut().enumerate() {
+            let standing_order_query = EntityQuery {
+                subgraph_id: SUBGRAPH_ID.clone(),
+                entity_types: vec!["StandingSellOrderBatch".to_string()],
+                filter: Some(
+                    EntityFilter::And(
+                        vec![
+                            EntityFilter::LessOrEqual("validFromAuctionId".to_string(), auction_id.to_value()),
+                            EntityFilter::Equal("accountId".to_string(), (reserved_account_id as u16).to_value())
+                        ]
+                    )
+                ),
+                order_by: Some(("batchIndex".to_string(), ValueType::BigInt)),
+                order_direction: Some(EntityOrder::Descending),
+                range: EntityRange {
+                    first: None,
+                    skip: 0,
+                },
+            };
+            let standing_order_option = self.reader
+                .find_one(standing_order_query)
+                .map_err(|e| DatabaseError::chain(ErrorKind::ConnectionError, "Could not execute query", e))?;
+
+            if let Some(standing_order) = standing_order_option {
+                let order_ids = standing_order.get("orders")
+                        .and_then(|orders| orders.clone().as_list())
+                        .ok_or_else(|| DatabaseError::new(ErrorKind::StateError, "No list orders found on standing order entity"))?;
+                    let relevant_order_query = entity_query("SellOrder", EntityFilter::In("id".to_string(), order_ids));
```

Comment:
Maybe we should capture here that we need to check that the orders are returned in the same sorting order as they were inserted. If not, we probably have to use some kind of other index to make sure they align with the order that the smart contract expects them to be in.

At the end of the day there should be a hash consistency check in the smart contract that ensures the orders are aligned correctly.

Reply:
Yes I agree. Should we make this an issue somewhere in one or both of the projects?